src/Compilers/Core/Portable/Collections/SmallConcurrentSetOfInts.cs

```diff
@@ -60,7 +62,7 @@ private static bool Contains(SmallConcurrentSetOfInts set, int i)
                     return true;
                 }
 
-                set = set._next;
+                set = set._next!;
```

Comment:
Among alternatives for this situation, my current preference is assigning the parameter to a new local:

```
SmallConcurrentSetOfInts? current = set;
do
{
  current = current._next;
}
while (current != null);
```
```

Reply:
@sharwell note, this is still in draft mode 😄